lib/lib.wrappers.php

```diff
@@ -71,3 +71,23 @@ function dj_escapeshellarg(?string $arg) : string
     if (!isset($arg) || $arg==='') return "''";
     return escapeshellarg($arg);
 }
+
+/**
+ * A simple function that allows to sleep for fractional seconds. Note that
+ * usleep is documented to consume CPU cycles and may not work for times
+ * larger than a second.
+ * Returns a boolean value for success or if the delay was interrupted by a
+ * signal, returns a float with the time remaining, similar to time_nanosleep.
+ */
+function dj_sleep(float $seconds) : bool
+{
+    $seconds_int = (int)$seconds;
+    $nanoseconds = (int)(1000000000*($seconds-$seconds_int));
+
+    $result = time_nanosleep($seconds_int, $nanoseconds);
+    if (is_array($result)) {
+        $result = $result['seconds'] + 0.000000001 * $result['nanoseconds'];
```

Comment:
Instead of the multiplication here, make the value above a constant and use division with the same constant?